allensdk/brain_observatory/ecephys/optotagging.py

```diff
@@ -0,0 +1,70 @@
+import pandas as pd
+import pynwb
+from pynwb import NWBFile
+
+from allensdk.brain_observatory.nwb import setup_table_for_epochs
+from allensdk.core import DataObject, JsonReadableInterface, \
+    NwbWritableInterface, NwbReadableInterface
+
+
+class OptotaggingTable(DataObject, JsonReadableInterface,
+                       NwbWritableInterface, NwbReadableInterface):
+    """Optotagging table - optotagging stimulation"""
+    def __init__(self, table: pd.DataFrame):
+        """
+
+        Parameters
+        ----------
+        table: A dataframe with columns:
+            - start_time: onset of stimulation
+            - condition: optical stimulation pattern
+            - level: intensity (in volts output to the LED) of stimulation
+            - stop_time: stop time of stimulation
+            - stimulus_name
+            - duration: duration of stimulation
+        """
+        super().__init__(name='optotaggging_table', value=table)
+
+    @classmethod
+    def from_json(cls, dict_repr: dict) -> "OptotaggingTable":
+        table = pd.read_csv(dict_repr['optotagging_table_path'])
+        return OptotaggingTable(table=table)
+
+    @classmethod
+    def from_nwb(cls, nwbfile: NWBFile) -> "OptotaggingTable":
+        mod = nwbfile.get_processing_module('optotagging')
+        table = mod.get_data_interface('optogenetic_stimulation')\
+            .to_dataframe()
+        table.drop(columns=['tags', 'timeseries'], inplace=True)
+        return OptotaggingTable(table=table)
+
+    def to_nwb(self, nwbfile: NWBFile) -> NWBFile:
+        optotagging_table = self.value
+
+        # "name" is a pynwb reserved column name that older versions of the
+        # pre-processed optotagging_table may use.
+        optotagging_table = \
```

Comment:
For the column rename here, doesn't this mean that the symmetry of `to_nwb` and `from_nwb` is broken? That is using `to_nwb` and loading with `from_nwb` won't result in the same data as the original class.

Reply:
We write the table with renamed columns to nwb. Then in `from_nwb` we just read in that table, so the column names are the same.

Reply:
You are right. If "name" was a column, it would be different between `from_json` and `from_nwb`. It turns out this column doesn't exist, which is why the test passes.

Reply:
@morriscb I moved this rename logic to the constructor so the tables will definitely be the same with different read* methods